src/dataflow/render/mod.rs

```diff
@@ -314,6 +319,61 @@ pub(crate) fn build_dataflow<A: Allocate>(
                             Envelope::Upsert(_) => unreachable!(),
                         };
 
+                        // Implement source filtering and projection.
```

Comment:
Take it or leave it, but `build_dataflow` is already pretty long/complicated... do you think this chunk of code would be better off as a separate function?

Reply:
My next plan is to rip all of this up and put it in separate places. I thought that might be better done with everyone on board though, and done consistently rather than one off. Happy to do it here, but there needs to be a follow-on that does it for everything else.